packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -123,6 +128,15 @@ export class ManagedKafkaEventSource extends StreamEventSource {
 
     target.role?.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaMSKExecutionRole'));
   }
+
+  private sourceAccessConfigurations() {
+    const sourceAccessConfigurations = [];
+    if (this.innerProps.secret !== undefined) {
+      sourceAccessConfigurations.push({ type: lambda.SourceAccessConfigurationType.SASL_SCRAM_512_AUTH, uri: this.innerProps.secret.secretArn });
+    }
+
+    return sourceAccessConfigurations;
```

Comment:
update this to return `undefined` if the list is empty

Reply:
Updated both MSK and self-managed.